### DISCUSSION TITLE
No

### DISCUSSION THREAD

**Discussion by majidam20** [alan-turing-institute/sktime#680] [Q&A] [answered]

**Comment by mloning** [comment, marked as answer]
Hi @majidam20, here are some thoughts:

* Whether or not the rows are [i.i.d.](https://en.wikipedia.org/wiki/Independent_and_identically_distributed_random_variables) is an assumption we make, which may be more or less plausible for the given data, and not determined by the shape of the data. Given that the rows represent data ordered in time, it's not very plausible to assume that they are independent: knowing past data will tell us something about future data, knowing the first row will tell us something about the next rows. 
* If, in the deployment situation, you observe up to xn when you want to predict yn, you can use the representation with this first row: x1, x2 … xn,yn. Otherwise, if you observe only up to xn-1  when you want to predict yn, you should use the representation with this first row: x1, x2 … xn,yn+1. It depends on *at which point* you have *what* data available in your deployment situation.

**Comment by mloning** [comment]
@majidam20 it's hard for me to follow what you're trying to do exactly or what your question is. Could you perhaps try to focus on a specific parts or post a minimal, reproducible piece of code?

**Comment by majidam20** [comment]


**Reply by mloning** [reply]
Shuffling in this case, from what I understand, seems fine as long as you properly evaluate your algorithm, so you need to make sure to evaluate your algorithm (that is the entire pipeline including the sliding-window transformation and oversampling) on future data.

**Comment by fkiraly** [comment]
Hm, sounds like your problem is predictive event modelling? Since your "anomaly" class is not really a class, but an event happening at a certain time. You can of course encode it as a class, but that's not necessarily the best way to approach this.

Event modelling is on the roadmap of sktime, but currently not supported.

**Reply by fkiraly** [reply]
@mloning FYI

**Comment by majidam20** [comment]


**Reply by mloning** [reply]
Sorry Majid, as said before, I still have a very hard time understanding what you're asking. Please post a [minimal, reproducible example](https://stackoverflow.com/help/mcve) of the problem you're trying to solve.

**Comment by fkiraly** [comment]
Hm, all questions got deleted? I assume it must be answered then...
